<html>
<head><script async src="https://www.googletagmanager.com/gtag/js?id=UA-83731338-2"></script><script>  window.dataLayer = window.dataLayer || [];  function gtag(){dataLayer.push(arguments);}  gtag('js', new Date());  gtag('config', 'UA-83731338-2');</script><title>11.5.1 The Classes Exception and RuntimeException</title>
<META NAME=MS-HAID CONTENT="11_5_1">
<style>@import url(stylesheets/msdn_ie4.css);</style>
<link disabled rel="stylesheet" href="stylesheets/msdn_ie3.css"></head>
<body bgcolor=#ffffff>

<h3>11.5.1	 The Classes Exception and RuntimeException</h3>
<a name="44285"></a>
<p>
<a name="44286"></a>
The class <code>Exception</code> is the superclass of all the exceptions that ordinary programs may wish to recover from.
<p><a name="44287"></a>
<h4>11.5.1.1	 Standard Runtime Exceptions</h4>
<p>
<a name="44288"></a>
The class <code>RuntimeException</code> is a subclass of class <code>Exception</code>. The subclasses 
of <code>RuntimeException</code> are unchecked exception classes.
<p><a name="44289"></a>
Package <code>java.lang</code> defines the following standard unchecked runtime exceptions, which, like all other classes in package <code>java.lang</code>, are implicitly imported and therefore may be referred to by their simple names:<p>
<ul><a name="44290"></a>
<li><code>ArithmeticException</code>: An exceptional arithmetic situation has arisen, such as an integer division <a href="15_16_2.htm#5047">(&#167;15.16.2)</a> operation with a zero divisor. 
<a name="44297"></a>
<li><code>ArrayStoreException</code>: An attempt has been made to store into an array component a value whose class is not assignment compatible with the component type of the array (<a href="10_10.htm#11430">&#167;10.10</a>, <a href="15_25_1.htm#5295">&#167;15.25.1</a>).
<a name="44304"></a>
<li><code>ClassCastException</code>: An attempt has been made to cast (<a href="5_4.htm#176921">&#167;5.4</a>, <a href="15_15.htm#238146">&#167;15.15</a>) a reference to an object to an inappropriate type.
<a name="44305"></a>
<li><code>IllegalArgumentException</code>: A method was passed an invalid or inappropriate argument or invoked on an inappropriate object. Subclasses of this class include:
<ul>
<a name="44306"></a>
<li><code>IllegalThreadStateException:</code> A thread was not in an appropriate state for a requested operation.
<a name="44307"></a>
<li><code>NumberFormatException</code>: An attempt was made to convert a <code>String</code> to a value of a numeric type, but the <code>String</code> did not have an appropriate format.
</ul>
<a name="44308"></a>
<li><code>IllegalMonitorStateException</code>: A thread has attempted to wait on (<a href="20_1.htm#33394">&#167;20.1.6</a>, <a href="20_1.htm#14926">&#167;20.1.7</a>, <a href="20_1.htm#32520">&#167;20.1.8</a>) or notify (<a href="20_1.htm#13789">&#167;20.1.9</a>, <a href="20_1.htm#13790">&#167;20.1.10</a>) other threads waiting on an object that it has not locked.
<a name="44315"></a>
<li><code>IndexOutOfBoundsException</code>: Either an index of some sort (such as to an array, a string, or a vector) or a subrange, specified either by two index values or by an index and a length, was out of range.
<a name="44316"></a>
<li><code>NegativeArraySizeException</code>: An attempt was made to create an array with a negative length <a href="15_9.htm#46168">(&#167;15.9)</a>.
<a name="44320"></a>
<li><code>NullPointerException</code>: An attempt was made to use a null reference in a case where an object reference was required.
<a name="44324"></a>
<li><code>SecurityException</code>: A security violation was detected <a href="20_17.htm#14464">(&#167;20.17)</a>.
</ul><p><a name="44325"></a>
Package <code>java.util</code> defines the following additional standard unchecked runtime 
exceptions:
<p><ul><a name="44326"></a>
<li><code>java.util.EmptyStackException</code>: An attempt was made to access an element of an empty stack.
<a name="44327"></a>
<li><code>java.util.NoSuchElementException</code>: An attempt was made to access an element of an empty vector.
</ul><p><a name="44328"></a>
<h4>11.5.1.2	 Standard Checked Exceptions</h4>
<p>
<a name="44329"></a>
The standard subclasses of <code>Exception</code> other than <code>RuntimeException</code> are all 
checked exception classes.
<p><a name="44330"></a>
Package <code>java.lang</code> defines the following standard exceptions, which, like all other classes in package <code>java.lang</code>, are implicitly imported and therefore may be referred to by their simple names:<p>
<ul><a name="44331"></a>
<li><code>ClassNotFoundException</code>: A class or interface with a specified name could not be found <a href="20_3.htm#28532">(&#167;20.3.8)</a>.
<a name="44338"></a>
<li><code>CloneNotSupportedException</code>: The <code>clone</code> method <a href="20_1.htm#14934">(&#167;20.1.5)</a> of class <code>Object</code> &#32;has been invoked to clone an object, but the class of that object does not implement the <code>Cloneable</code> interface.
<a name="44339"></a>
<li><code>IllegalAccessException</code>: An attempt has been made to load a class using a string giving its fully qualified name, but the currently executing method does not have access to the definition of the specified class because the class is not <code>public</code> and is in another package.
<a name="44340"></a>
<li><code>InstantiationException</code>: An attempt was made to create an instance of a class using the <code>newInstance</code> method in class <code>Class</code>, but the specified class object cannot be instantiated because it is an interface, is <code>abstract</code>, or is an array.
<a name="44341"></a>
<li><code>InterruptedException</code>: The current thread was waiting, and another thread has interrupted the current thread, using the <code>interrupt</code> method of class <code>Thread</code> <a href="20_20.htm#29179">(&#167;20.20.31)</a>.
</ul><p><a name="44345"></a>
Package <code>java.io</code> defines the following additional standard exceptions:
<p><ul><a name="44346"></a>
<li><code>java.io.IOException</code>: A requested I/O operation could not be completed normally. Subclasses of this class include:
<ul>
<a name="44347"></a>
<li><code>java.io.EOFException</code>: End of file has been encountered before normal completion of an input operation.
<a name="44348"></a>
<li><code>java.io.FileNotFoundException</code>: A file with the name specified by a file name string or path was not found within the file system.
<a name="44349"></a>
<li><code>java.io.InterruptedIOException</code>: The current thread was waiting for completion of an I/O operation, and another thread has interrupted the current thread, using the <code>interrupt</code> method of class <code>Thread</code> <a href="20_20.htm#29179">(&#167;20.20.31)</a>.
<a name="44353"></a>
<li><code>java.io.UTFDataFormatException</code>: A requested conversion of a string to or from Java modified UTF-8 format could not be completed (<a href="22_1.htm#28916">&#167;22.1.15</a>, <a href="22_2.htm#29074">&#167;22.2.14</a>) because the string was too long or because the purported UTF-8 data was not the result of encoding a Unicode string into UTF-8.
</ul>
</ul><p><a name="44360"></a>
The standard package <code>java.net</code> defines the following additional subclasses of 
<code>java.io.IOException</code>:
<p><a name="44361"></a>
u	 <code>java.net.MalformedURLException</code>: A string that was provided as a URL, or as part of a URL, had an inappropriate format or specified an unknown protocol.<p>
<ul><ul>
<a name="44362"></a>
<li><code>java.net.ProtocolException</code>: Some aspect of a network protocol was not correctly carried out.
<a name="44363"></a>
<li><code>java.net.SocketException</code>: An operation involving a socket could not be completed normally.
<a name="44364"></a>
<li><code>java.net.UnknownHostException</code>: The name of a network host could not be resolved to a network address.
<a name="44365"></a>
<li><code>java.net.UnknownServiceException</code>: The network connection cannot support the requested service.
</ul>
</ul><p>

</body></html>
